syntax = "proto3";

enum TaskType {
  DEFAULT = 0;
  ADD = 1;
  SUBTRACT = 2;
  MULTIPLY = 3;
  DIVIDE = 4;
}

enum TaskStatus {
  UNKNOWN = 0;
  QUEUED = 1;
  RUNNING = 2;
  DONE = 3;
  FAILED = 4;
}

message Task {
  string id = 1; // Task identifier
  TaskType type = 2; // Type of operation
  double operand1 = 3; // First operand
  double operand2 = 4; // Second operand
  int32 timeout = 5; // Timeout in seconds
  int32 processing_time = 6; // Simulated execution time in seconds
}

message TaskResult {
  string id = 1; // Task identifier
  TaskStatus status = 2; // Status of the task
  string description = 3; // Optional description eg. for error
  double result = 4; // Optional result of the task
}

message WorkerProperties {
  string id = 1; // Worker identifier
  repeated TaskType capabilities = 2; // Capabilities of the worker
}
